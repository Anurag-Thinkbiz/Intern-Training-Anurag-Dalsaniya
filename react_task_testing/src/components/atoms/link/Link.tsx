import { linkType } from "../../../data/modal/types/hookTypes/hookType";
import { StyledLink } from "../../styles/button.style";

const Link = ({ src, text }: linkType) => {
  return <StyledLink href={src}>{text}</StyledLink>;
};
export default Link;
