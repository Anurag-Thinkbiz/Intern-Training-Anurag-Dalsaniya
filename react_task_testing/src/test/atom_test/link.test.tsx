import { render } from "@testing-library/react";
import Link from "../../components/atoms/link/Link";
jest.mock("../../components/styles/button.style", () => ({
  StyledLink: ({ children, ...props }: any) => <a {...props}>{children}</a>,
}));

describe("Link Component Snapshot", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <Link src="https://example.com" text="Click Here" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
