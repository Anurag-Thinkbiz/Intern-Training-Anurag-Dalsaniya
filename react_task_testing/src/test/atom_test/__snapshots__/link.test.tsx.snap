// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Component Snapshot renders correctly 1`] = `
<DocumentFragment>
  <a
    href="https://example.com"
  >
    Click Here
  </a>
</DocumentFragment>
`;
